e2e/_suites/ingest-manager/ingest-manager_test.go

```diff
@@ -0,0 +1,208 @@
+	s.Step(`^there is no data in the index$`, imts.thereIsNoDataInTheIndex)
+	s.Step(`^an agent is enrolled$`, imts.anAgentIsEnrolled)
+	s.Step(`^the agent is re-enrolled on the host$`, imts.theAgentIsReenrolledOnTheHost)
+	s.Step(`^the enrollment token is revoked$`, imts.theEnrollmentTokenIsRevoked)
+	s.Step(`^an attempt to enroll a new agent fails$`, imts.anAttemptToEnrollANewAgentFails)
+
+	s.BeforeSuite(func() {
+		log.Debug("Installing ingest-manager runtime dependencies")
+		serviceManager := services.NewServiceManager()
+
+		env := map[string]string{
+			"stackVersion": stackVersion,
+		}
+
+		profile := "ingest-manager"
+		err := serviceManager.RunCompose(true, []string{profile}, env)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"profile": profile,
+			}).Error("Could not run the runtime dependencies for the profile.")
+		}
+
+		minutesToBeHealthy := 3 * time.Minute
+		healthy, err := e2e.WaitForElasticsearch(minutesToBeHealthy)
+		if !healthy {
+			log.WithFields(log.Fields{
+				"error":   err,
+				"minutes": minutesToBeHealthy,
+			}).Error("The Elasticsearch cluster could not get the healthy status")
+		}
+
+		healthyKibana, err := e2e.WaitForKibana(minutesToBeHealthy)
+		if !healthyKibana {
+			log.WithFields(log.Fields{
+				"error":   err,
+				"minutes": minutesToBeHealthy,
+			}).Error("The Kibana instance could not get the healthy status")
+		}
+	})
+	s.BeforeScenario(func(*messages.Pickle) {
+		log.Debug("Before Ingest Manager scenario")
+	})
+	s.AfterSuite(func() {
+		log.Debug("Destroying ingest-manager runtime dependencies")
+		serviceManager := services.NewServiceManager()
+		profile := "ingest-manager"
+
+		err := serviceManager.StopCompose(true, []string{profile})
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":   err,
+				"profile": profile,
+			}).Warn("Could not destroy the runtime dependencies for the profile.")
+		}
+	})
+	s.AfterScenario(func(*messages.Pickle, error) {
+		log.Debug("After Ingest Manager scenario")
+	})
+}
+
+// IngestManagerTestSuite represents a test suite, holding references to the pieces needed to run the tests
+type IngestManagerTestSuite struct {
+}
+
+func (imts *IngestManagerTestSuite) kibanaSetupHasBeenExecuted(setup string) error {
+	log.WithFields(log.Fields{
+		"setup": setup,
+	}).Debug("Creating Kibana setup")
+
+	type payload struct {
+		ForceRecreate bool `json:"forceRecreate"`
+	}
+
+	data := payload{
+		ForceRecreate: true,
+	}
+	payloadBytes, err := json.Marshal(data)
+	if err != nil {
+		log.Error("Could not serialise payload")
+		return err
+	}
+
+	// running on localhost as Kibana is expected to be exposed there
+	fleetSetupURL := "http://localhost:5601/api/ingest_manager/fleet/setup"
+	err = curl.Post(fleetSetupURL, payloadBytes)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+			"url":   fleetSetupURL,
+		}).Error("Could not initialise Fleet")
+		return err
+	}
+
+	log.Debug("Ensuring Fleet was initialised")
```

Comment:
I'd like to check that Fleet was configured properly: is there a check (on Kibana API or elasticsearch) to create an assertion for this?

Reply:
I believe you can use the `get-fleet-setup` route (`GET /api/ingest_manager/fleet/setup`) which ~will return an object like `{ isInitialized: true }` or `{ isInitialized: false }`~

~From the OpenAPI spec file
https://github.com/elastic/kibana/blob/974dcab91733ff7adeabb760e0373d5bbcbfb1cf/x-pack/plugins/ingest_manager/common/openapi/spec_oas3.json#L1417-L1459~

The spec file is incorrect. I believe the actual response shape is 

```typescript
{
  isReady: boolean;
  missing_requirements: Array<'tls_required' | 'api_keys' | 'fleet_admin_user'>;
}
```

From https://github.com/elastic/kibana/blob/5e972e14d147e3c5232114690f61a96a493a28ae/x-pack/plugins/ingest_manager/common/types/rest_spec/fleet_setup.ts#L11-L15

Reply:
The POST request for creating the setup returns `isInitialized: true`, so I guess this could be enough